core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.spec.ts

```diff
@@ -1156,6 +1157,203 @@ describe('Contributions and review component', () => {
+              id: '1',
+              question_state_data: {
+                content: {
+                  html: 'Question 2',
+                  content_id: 'content_2'
+                },
+                interaction: {
+                  answer_groups: [{
+                    outcome: {
+                      dest: 'outcome 1',
+                      dest_if_really_stuck: null,
+                      feedback: {
+                        content_id: 'content_1',
+                        html: ''
+                      },
+                      labelled_as_correct: true,
+                      param_changes: [],
+                      refresher_exploration_id: null
+                    },
+                    rule_specs: [],
+                  }],
+                  confirmed_unclassified_answers: [],
+                  customization_args: {
+                    placeholder: {
+                      value: {
+                        content_id: 'ca_placeholder_0',
+                        unicode_str: ''
+                      }
+                    },
+                    rows: { value: 1 },
+                    catchMisspellings: {
+                      value: false
+                    }
+                  },
+                  default_outcome: {
+                    dest: null,
+                    dest_if_really_stuck: null,
+                    feedback: {
+                      html: 'Correct Answer',
+                      content_id: 'content_2'
+                    },
+                    param_changes: [],
+                    labelled_as_correct: true
+                  },
+                  hints: [{
+                    hint_content: {
+                      html: 'Hint 1',
+                      content_id: 'content_3'
+                    }
+                  }],
+                  solution: {
+                    correct_answer: 'This is the correct answer',
+                    answer_is_exclusive: false,
+                    explanation: {
+                      html: 'Solution explanation',
+                      content_id: 'content_4'
+                    }
+                  },
+                  id: 'TextInput'
+                },
+                param_changes: [],
+                recorded_voiceovers: {
+                  voiceovers_mapping: {}
+                }
+              },
+            }
+          },
+          status: 'accepted'
+        };
+
+        // Now load a question contribution.
+        getUserCreatedQuestionSuggestionsAsyncSpy.and.returnValue(
+          Promise.resolve({
+            suggestionIdToDetails: {
+              suggestion_2: {
+                suggestion: suggestion2,
+                details: {
+                  skill_id: 'skill_1',
+                  skill_description: 'skill_1'
+                }
+              }
+            },
+            more: false
+          }));
+
+        component.switchToTab(
+          component.TAB_TYPE_CONTRIBUTIONS, 'add_question');
```

Comment:
Do we need to execute `switchToTab` twice for testing "Should show only selected type when switching tabs"

Reply:
@gp201  thanks for the review. The aim of this test is to prove that when loadContributions is called, the only contributions loaded are for the selected tab. 

Previously, because questions and translations held in the contributions array use different indexes, when switching between tabs, the contributions array could still contain data from the old tab. This causes problems in the ui later.   

So I think we should leave the 2 calls to switchTo Tab in, to try and make it obvious what we are testing, but perhaps I should add some comments to explain better what the test is doing?  PTAL

Reply:
@stevenjgodfrey that makes sense. Thanks for the explanation. I think it would be better to add comments to explain what the test is doing as you suggested.

Reply:
Done